static/js/public/first-snap-flow.js

```diff
@@ -255,22 +257,66 @@ function initRegisterName(formEl, notificationEl, successEl) {
   formEl.addEventListener("submit", event => {
     event.preventDefault();
     let formData = new FormData(formEl);
+    const submitButton = formEl.querySelector(
+      "[data-js='js-snap-name-register']"
+    );
+
+    const currentPanel = formEl.closest(".p-accordion__group");
+    const currentToggle = currentPanel.querySelector(".p-accordion__tab");
+    const nextPanel = currentPanel.nextElementSibling;
+    const nextToggle = nextPanel.querySelector(".p-accordion__tab");
+
+    // Show spinner if data fetch takes long
+    const timer = setTimeout(() => {
+      if (currentToggle.getAttribute("aria-expanded") === "true") {
+        submitButton.classList.add("has-spinner");
+        submitButton.disabled = true;
+      }
+    }, 400);
 
     fetch("/register-snap/json", {
       method: "POST",
       body: formData
     })
       .then(response => response.json())
       .then(json => {
+        clearTimeout(timer);
         if (json.errors) {
           showError(json.errors[0].message);
+          if (submitButton.disabled) {
```

Comment:
This is repeated a couple of times, could be a function above the fetch somewhere

```
const enableButton = () => {
  if (submitButton.disabled) {
    submitButton.classList.remove("has-spinner");
    submitButton.disabled = false;
  }
}
```

Then just call `enableButton()` in each of these if's. That way if we want to add a different class in the future we just update 1 place.